src/test/javascript/spec/e2e.ts

```diff
@@ -78,3 +100,118 @@ describe('Screenshot Tests', () => {
+
+    await geneticTypeSelect.click();
+
+    const firstGeneticTypeSelectOption = await $(`div[id="react-select-${DATA_IMPORT_GENETIC_TYPE_SELECT_ID}-option-0"]`);
+    expect(firstGeneticTypeSelectOption).toExist();
+    await firstGeneticTypeSelectOption.click();
+
+    await dataTypeSelect.click();
+
+    // this select is grouped, therefore the pattern 0-0
+    const firstDataTypeSelectOption = await $(`div[id="react-select-${DATA_IMPORT_DATA_TYPE_SELECT_ID}-option-0-0"]`);
+    expect(firstDataTypeSelectOption).toExist();
+    await firstDataTypeSelectOption.click();
+
+    const fileFormatWarningAlert = await $(`div[id=${DATA_IMPORT_FILE_FORMAT_WARNING_ALERT_ID}]`);
+    expect(fileFormatWarningAlert).toExist();
+  });
+
+  const selectGenomicIndicatorToImport = async () => {
+    const geneticTypeSelect = await $(`div[id="${DATA_IMPORT_GENETIC_TYPE_SELECT_ID}"]`);
+    const dataTypeSelect = await $(`div[id="${DATA_IMPORT_DATA_TYPE_SELECT_ID}"]`);
+
+    await geneticTypeSelect.click();
+
+    const germlineOption = await geneticTypeSelect.$('aria/Germline');
+    expect(germlineOption).toExist();
+    await germlineOption.click();
+
+    await dataTypeSelect.click();
+
+    // genomic indicator option has both required and optional columns
+    const genomicIndicatorOption = await dataTypeSelect.$('aria/Genomic Indicator');
+    expect(genomicIndicatorOption).toExist();
+    await genomicIndicatorOption.click();
+  };
+
+  const uploadGenomicIndicatorToImport = async () => {
+    await selectGenomicIndicatorToImport();
+
+    const fileToUpload = path.join(process.cwd(), 'src/test/javascript/data/genomic_indicators.tsv');
+    const remoteFilePath = await browser.uploadFile(fileToUpload);
+    await $(`input[id=${DATA_IMPORT_FILE_UPLOAD_INPUT_ID}]`).setValue(remoteFilePath);
+  };
+
+  it('File required and optional columns alerts are shown', async () => {
+    await selectGenomicIndicatorToImport();
+
+    const requiredColumnsAlert = await $(`div[id=${DATA_IMPORT_REQUIRED_COLUMNS_INFO_ALERT_ID}]`);
+    expect(requiredColumnsAlert).toExist();
+
+    const optionalColumnsAlert = await $(`div[id=${DATA_IMPORT_OPTIONAL_COLUMNS_INFO_ALERT_ID}]`);
+    expect(optionalColumnsAlert).toExist();
+  });
+
+  it('Properly process genomic indicator file', async () => {
+    await uploadGenomicIndicatorToImport();
+
+    // expand sidebar to take screenshot
+    const expandIcon = await $(`span[data-testid="${EXPAND_SIDEBAR_BUTTON_ID}"]`);
+    await expandIcon.click();
+
+    // too complicated to check logics, do a screenshot instead
+    const dataTable = await $(`div[id=${DATA_IMPORT_DATA_TABLE_ID}]`);
+    expect(dataTable).toExist();
+    const result = await browser.checkElement(dataTable, 'data-import-data-table-before-import', methodOptions);
+    expect(result).toBeLessThanOrEqual(0);
+  });
+
+  it('Data properly imported and show warning/error if available', async () => {
```

Comment:
I am a little more hesitant to add this one to screenshot tests since it checks if the import button is disabled. However, if it needs to save a screenshot I am leaning toward putting it in screenshot tests.

Reply:
Agree